src/command.ts

```diff
@@ -75,6 +77,66 @@ export class Command<C extends Context = Context> implements MiddlewareObj<C> {
         this._languages.set("default", { name: name, description });
     }
 
+    /**
+     * Whether the command has a custom prefix
+     */
+    get hasCustomPrefix() {
+        return this.prefix && this.prefix !== "/";
+    }
+
+    /**
+     * Gets the command name as string
+     */
+    public get stringName() {
+        return typeof this.name === "string" ? this.name : this.name.source;
+    }
+
+    /**
+     * Whether the command can be passed to a `setMyCommands` API call
+     * and, if not, the reason.
+     */
+    public isApiCompliant(
+        language?: LanguageCode | "default",
+    ): [result: true] | [
+        result: false,
+        reason: string,
+    ] {
+        if (this.hasCustomPrefix) {
+            return [
+                false,
+                `Command has custom prefix: ${this._options.prefix}`,
+            ];
+        }
+
+        const name = language ? this.getLocalizedName(language) : this.name;
+
+        if (typeof name !== "string") {
+            return [false, "Command has a regular expression name"];
+        }
+
+        if (name.toLowerCase() !== name) {
+            return [false, "Command name has uppercase characters"];
+        }
+
+        if (name.length > 32) {
+            return [
+                false,
+                `Command name is too long (${name.length} characters). Maximum allowed is 32 characters`,
+            ];
+        }
+
+        if (!COMMAND_NAME_REGEX.test(name)) {
+            return [
+                false,
+                `Command name has special characters (${
+                    name.replace(/[0-9a-z_]/g, "")
+                }). Only lowercase letters, digits and _ are allowed`,
+            ];
+        }
+
```

Comment:
A command can be uncompliant for more than one reason at the same time. for example, the custom prefixed command: `$SUPERuncompli4ntCommand12345678` violates every rule.

I think it would be good to stack / list every uncompliance at once. So if someone has an uncompliant command and is willing to correct it, can know at first try all the reason to do so.

Reply:
you're correct
I'll change that

Reply:
this is addressed on the latest commit :)